internal/pkg/consul/client.go

```diff
@@ -0,0 +1,385 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package consul
+
+import (
+	"fmt"
+	"github.com/edgexfoundry/go-mod-registry"
+	consulapi "github.com/hashicorp/consul/api"
+	"github.com/mitchellh/consulstructure"
+	"github.com/pelletier/go-toml"
+	"github.com/pkg/errors"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const consulStatusPath = "/v1/agent/self"
+const serviceStatusPass = "passing"
+
+
+type consulClient struct {
+	consulUrl           string
+	consulClient        *consulapi.Client
+	consulConfig        *consulapi.Config
+	configBasePath      string
+	serviceId           string
+	serviceAddress      string
+	servicePort         int
+	healthCheckUrl      string
+	healthCheckInterval string
+}
+
+// Create new Consul Client. Service details are optional, not needed just for configuration, but required if registering
+func NewConsulClient(registryConfig registry.Config, serviceId string) (*consulClient, error) {
+
```

Comment:
Is there going to be some equivalent function as the one on the line below?
https://github.com/edgexfoundry/edgex-go/blob/1c12c2fe67b376e4e8d590974ed9fd1a5305ddf7/internal/core/data/init.go#L164

The current ConsulConfig (registry.Config equivalent) is built from several discrete pieces of the overall ConfigurationStruct from each service. Should this be in this module or a function within edgex-go?

Reply:
The calling service builds registry.Config from those discrete pieces. See example in README.md